PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ["stock.move"]
-    _inherit = ["stock.move.line"]
-    _inherit = ["stock.picking.type"]
-    _inherit = ["stock.picking"]
-    _inherit = ['stock.warehouse']

Total Changes: 11

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock_picking_batch/models/stock_move.py
+++ b/addons/stock_picking_batch/models/stock_move.py
@@ -6,7 +6,7 @@ from odoo.osv import expression
 
 
 class StockMove(models.Model):
-    _inherit = ["stock.move"]
+    _inherit = "stock.move"
 
     def _search_picking_for_assignation_domain(self):
         domain = super()._search_picking_for_assignation_domain()

--- a/addons/stock_picking_batch/models/stock_move_line.py
+++ b/addons/stock_picking_batch/models/stock_move_line.py
@@ -10,7 +10,7 @@ from odoo.tools.misc import OrderedSet
 
 
 class StockMoveLine(models.Model):
-    _inherit = ["stock.move.line"]
+    _inherit = "stock.move.line"
 
     batch_id = fields.Many2one(related='picking_id.batch_id')
 

--- a/addons/stock_picking_batch/models/stock_picking.py
+++ b/addons/stock_picking_batch/models/stock_picking.py
@@ -7,7 +7,7 @@ from odoo.exceptions import ValidationError
 
 
 class StockPickingType(models.Model):
-    _inherit = ["stock.picking.type"]
+    _inherit = "stock.picking.type"
 
     count_picking_batch = fields.Integer(compute='_compute_picking_count')
     count_picking_wave = fields.Integer(compute='_compute_picking_count')
@@ -84,7 +84,7 @@ class StockPickingType(models.Model):
 
 
 class StockPicking(models.Model):
-    _inherit = ["stock.picking"]
+    _inherit = "stock.picking"
 
     batch_id = fields.Many2one(
         'stock.picking.batch', string='Batch Transfer',

--- a/addons/stock_picking_batch/models/stock_picking_batch.py
+++ b/addons/stock_picking_batch/models/stock_picking_batch.py
@@ -10,6 +10,7 @@ from odoo.tools import float_is_zero, format_list
 
 
 class StockPickingBatch(models.Model):
+    _name = 'stock.picking.batch'
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _description = "Batch Transfer"
     _order = "name desc"

--- a/addons/stock_picking_batch/models/stock_warehouse.py
+++ b/addons/stock_picking_batch/models/stock_warehouse.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockWarehouse(models.Model):
-    _inherit = ['stock.warehouse']
+    _inherit = 'stock.warehouse'
 
     def _get_picking_type_create_values(self, max_sequence):
         data, next_sequence = super()._get_picking_type_create_values(max_sequence)
